src/map/lua/lua_baseentity.cpp

```diff
@@ -2142,6 +2142,55 @@ inline int32 CLuaBaseEntity::hideNPC(lua_State *L)
     return 0;
 }
 
+/************************************************************************
+*  Function: despawnNPC()
+*  Purpose : despawns an npc right away and removes any queued actions.
+*  Example : npc:despawnNPC()
+*  Notes   : Currently only used to force despawn caskets.
+************************************************************************/
+
+inline int32 CLuaBaseEntity::despawnNPC(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_NPC);
+
+    if (m_PBaseEntity->status == STATUS_NORMAL)
+    {
+        m_PBaseEntity->status = STATUS_DISAPPEAR;
+        m_PBaseEntity->loc.zone->PushPacket(m_PBaseEntity, CHAR_INRANGE, new CEntityUpdatePacket(m_PBaseEntity, ENTITY_DESPAWN, UPDATE_COMBAT));
+        m_PBaseEntity->PAI->QueueEmpty();
+    }
+    return 0;
+}
+
+/************************************************************************
+*  Function: releaseAllFromNPC()
+*  Purpose : releases any players currently interacting with the npc
+*  Example : npc:releaseAllFromNPC()
```

Comment:
There is no event release or skip message for it if the NPC depsawns in most cases. Retail generally will leave you sitting in the menu, but then any selection you make doesn't execute (you can go through the motions but it won't give you any item).  In the case of caskets in FoV zones I've left the menu open for a literal hour before when I afk'd without realizing I didn't get enter. That was in Batallia Downs. The function might still be useful for other cases though.

Reply:
yeah i must admit i wasn't going to go down this route at first but i think theres a bug with caskets that mess the player up if no release packet is sent, ive had a few people say that when they have been testing it, sometimes they get "unable to attack target" and zoning into some random zone, im not 100% sure why but this seems to help with the issue, but if you think i should not use it then thats fine i can just remove it from the caskets.lua

Reply:
i do already have checks to see if the casket is spawned or not after an option is chosen so its not going to matter too much